packages/player/src/machines/video-machine.ts

```diff
@@ -51,262 +71,402 @@ export interface VideoStateContext {
         },
-        END_SEEKING: {
-          actions: [
-            assign({
-              seekingTime: (context, event) => 0,
-              seeking: (_context, _event) => false,
-            }),
-          ],
+      },
+      ready: {
+        initial: 'paused',
+        entry: [
+          assign({
+            waiting: (_context, _event) => false,
+            isSubmittingCueNote: (_context, _event) => false,
+          }),
+        ],
+        on: {
+          TAKE_NOTE: {
+            target: 'taking_note',
+          },
+          VOLUME_CHANGE: {
+            actions: [
+              assign({
+                volume: (_context, event) => event.volume,
+              }),
+              'setVolume',
+            ],
+          },
+
+          TOGGLE_MUTE: {
+            actions: ['toggleMute'],
+          },
+          PLAYBACKRATE_CHANGE: {
+            actions: [
+              assign({
+                playbackRate: (_context, event) => event.playbackRate,
+              }),
+              'setPlaybackRate',
+            ],
+          },
+        },
+        states: {
+          paused: {
+            on: {
+              PLAY: {
+                target: 'playing',
+                actions: [
+                  assign({
+                    hasStarted: (_context, _event) => true,
+                    lastAction: (_context, _event) => 'PLAY',
+                  }),
+                  'playVideo',
+                ],
+              },
+            },
+          },
+          playing: {
+            entry: [
+              assign({
+                waiting: (_context, _event) => false,
+              }),
+            ],
+            on: {
+              PAUSE: {
+                target: 'paused',
+                actions: [
+                  'pauseVideo',
+                  assign({
+                    lastAction: (_context, _event) => 'PAUSE',
+                  }),
+                ],
+              },
+              TIMING: {
+                actions: assign({
+                  currentTime: (context, _event) =>
+                    context.videoRef?.current?.currentTime ?? 0,
+                  duration: (context, _event) =>
+                    context.videoRef?.current?.duration ?? 0,
+                }),
+              },
+              END: 'ended',
+            },
+          },
+          ended: {
+            entry: ['onVideoEnded'],
+            on: {PLAY: 'playing'},
+          },
         },
       },
-      states: {
-        paused: {
-          on: {
-            PLAY: {
-              target: 'playing',
+      taking_note: {
+        initial: 'focused',
+        on: {
+          SUBMITTED: {
+            target: 'taking_note.submitting',
+            actions: [
+              assign({
+                isSubmittingCueNote: (_context, _event) => true,
+                metadataTracks: (context, event) => {
+                  context.metadataTracks[0].addCue(event.cue)
+                  return context.metadataTracks
```

Comment:
we can simply `addCue` to current notes metadata track which visually updates them right away and then continue on making request to a server.